### ISSUE TITLE
PCRE2 (PHP>=7.3) conditional substitution not working in real life

### ISSUE THREAD

**Issue by CarloMoretti-Repa** [firasdib/Regex101#2308]
## Bug Description
PCRE2 (PHP>=7.3) conditional substitution not working in real life.
The regex substitution in https://regex101.com/r/DIbRPF/1 won't work in real PHP code

### Reproduction steps
<?php
$re = '/^([0-9]{3})-?([0-9]{3})-?([0-9]{3})-?([0-9]{3})?$/mi';
$str = "
009026405
226-356-839
000147765015
";
$subst = '[$1.$2.$3.${4:+$4:000}]';
$result = preg_replace($re, $subst, $str);
echo "\nresult: \n".$result;

### Expected Outcome
[009.026.405.000]
[226.356.839.000]
[000.147.765.015]

actual output:
[009.026.405.${4:+:000}]
[226.356.839.${4:+:000}]
[000.147.765.${4:+015:000}]

### Browser
Firefox 127.01 64bit

### OS
1. PHP 8.1.2-1ubuntu2.18 (cli) (built: Jun 14 2024 15:52:55) on Linux 6.5.0-41-generic #41~22.04.2-Ubuntu SMP PREEMPT_DYNAMIC Mon Jun  3 11:32:55 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
2. PHP 7.4.3-4ubuntu2.23 (cli) (built: Jun 17 2024 13:22:20) ( NTS ) on Linux 5.4.0-182-generic #202-Ubuntu SMP Fri Apr 26 12:29:36 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

**Comment by scrutinizer11** [comment]
I use a regex editor powered by PCRE 8.xx (the old nomenclature, pre-PCRE 2021). The editor was unable to recognize the syntax you used. It seems applications are yet to maintain an equal footing regarding the standardized implementation of the PCRE protocol.

This code returned a success: 

Search for 
```bash
^([0-9]{3})-?([0-9]{3})-?([0-9]{3})-?([0-9]{3})?(?(4))$
``` 
replacing with 
```bash
$1.$2.$3.000
```

**Comment by CarloMoretti-Repa** [comment]
i think the first version of PCRE2 is the one called PCRE2-10.00 (see [here](https://github.com/PCRE2Project/pcre2/releases/tag/pcre2-10.00)), i guess the PCRE 8.xx editor won't recognize conditional substitution that has been introduced in PCRE2-10.00
Back to the main issue: regex101 claims that PHP 7.3+ is able to use PCRE2 conditional substitution, but it is not.

**Comment by working-name** [comment]
@CarloMoretti-Repa 

I think I see where the confusion lies. The `PHP >=7.3` is an example of something that implements PCRE2. It doesn't mean that the regex flavor **_is PHP_**. 

According to [pcre2.txt](https://www.pcre.org/pcre2.txt), the conditional substitution exists, and it's turned on via `PCRE2_SUBSTITUTE_EXTENDED`.

Here's an excerpt:

```py
 The second effect of setting PCRE2_SUBSTITUTE_EXTENDED is to  add  more
       flexibility  to  capture  group  substitution. The syntax is similar to
       that used by Bash:

         ${<n>:-<string>}
         ${<n>:+<string1>:<string2>}

       As before, <n> may be a group number or a name. The first  form  speci-
       fies  a  default  value. If group <n> is set, its value is inserted; if
       not, <string> is expanded and the  result  inserted.  The  second  form
       specifies  strings that are expanded and inserted when group <n> is set
       or unset, respectively. The first form is just a  convenient  shorthand
       for

         ${<n>:+${<n>}:<string>}

       Backslash  can  be  used to escape colons and closing curly brackets in
       the replacement strings. A change of the case forcing  state  within  a
       replacement  string  remains  in  force  afterwards,  as  shown in this
       pcre2test example:

         /(some)?(body)/substitute_extended,replace=${1:+\U:\L}HeLLo
             body
          1: hello
             somebody
          1: HELLO
```

So I guess then the question is - what implements PCRE2 extended substitution? 🤷‍♂️

**Comment by CarloMoretti-Repa** [comment]
> The PHP >=7.3 is an example of something that implements PCRE2

ok, so i think that there should be a clearer message somewhere, like "PCRE2 are **partially** implemented in PHP >= 7.3, **conditional substitution EXCLUDED**".
thanks.
